code/modules/research/destructive_analyzer.dm

```diff
@@ -52,9 +45,91 @@ Note: Must be placed within 3 tiles of the R&D Console
 		flick("d_analyzer_la", src)
 		addtimer(CALLBACK(src, .proc/finish_loading), 10)
 
-/obj/machinery/r_n_d/destructive_analyzer/proc/finish_loading()
+/obj/machinery/rnd/destructive_analyzer/proc/finish_loading()
 	update_icon()
-	busy = FALSE
+	reset_busy()
+
+/obj/machinery/rnd/destructive_analyzer/update_icon()
+	if(loaded_item)
+		icon_state = "d_analyzer_l"
+	else
+		icon_state = initial(icon_state)
+
+/obj/machinery/rnd/destructive_analyzer/proc/reclaim_materials_from(obj/item/thing)
+	. = 0
+	if(linked_console && linked_console.linked_lathe) //Also sends salvaged materials to a linked protolathe, if any.
+		for(var/material in thing.materials)
+			var/can_insert = min((linked_console.linked_lathe.materials.max_amount - linked_console.linked_lathe.materials.total_amount), (max(thing.materials[material]*(decon_mod/10), thing.materials[material])))
+			linked_console.linked_lathe.materials.insert_amount(can_insert, material)
+			. += can_insert
+
+/obj/machinery/rnd/destructive_analyzer/proc/destroy_item(obj/item/thing, innermode = FALSE)
+	if(QDELETED(thing) || QDELETED(src) || QDELETED(linked_console))
+		return FALSE
+	if(!innermode)
+		flick("d_analyzer_process", src)
+		busy = TRUE
+		addtimer(CALLBACK(src, .proc/reset_busy), 24)
+		use_power(250)
+		if(thing == loaded_item)
+			loaded_item = null
+		update_icon()
+		var/list/food = thing.GetAllContents() - thing
+		for(var/obj/item/innerthing in food)
+			destroy_item(innerthing, TRUE)

```

Comment:
This really should be handled by the item, not the DA

Reply:
what do you mean?